uaclient/clouds/aws.py

```diff
@@ -18,34 +21,60 @@
 class UAAutoAttachAWSInstance(AutoAttachCloudInstance):
 
     _api_token = None
+    _ip_addresss_type = None
```

Comment:
minor spelling (addresss), but even better I think we don't need to use _ip_address_type and instead just use _ip_address, because we are always looking up the value in the IMDS_IP_ADDRESS in all cases anyway. No need to use a dict to represent this content as we don't "need" ipv4 vs ipv6 keys for any of our processing.

```suggestion
    _ip_address = None
```